Documentation/CodingGuidelines/CodingBestPractices/NamedArguments.rst

```diff
@@ -0,0 +1,247 @@
+avoid this scenario.
+
+Invoking core API
+-----------------
+
+Within the TYPO3 core, named arguments are not used when invoking its own methods.
+There are exceptions in specific scenarios as outlined below, however these are
+the reasons for not using named arguments:
+
+* TYPO3 core tries to be as consistent as possible
+
+* Setting a good example for extensions
+
+* Avoiding complications and side effects during refactoring
+
+* Addressing legacy code within the TYPO3 core containing methods with less-desirable
+  variable names, aiming for gradual improvement without disruptions
+
+* Preventing issues with inheritance, especially in situations like this:
+
+  .. code-block:: php
+      :caption: PHP error using named arguments and inheritance
+
+      interface I {
+          public function test($foo, $bar);
+      }
+
+      class C implements I {
+          public function test($a, $b) {}
+      }
+
+      $obj = new C();
+
+      // Pass params according to I::test() contract
+      $obj->test(foo: "foo", bar: "bar"); // ERROR!
+
+Utilizing Named Arguments in PHPUnit Test Data Providers
+--------------------------------------------------------
+
+The use of named arguments in PHPUnit test data providers is permitted and
+encouraged, particularly when enhancing readability. Take, for example, this
+instance where PHPUnit utilizes the array keys :php:`languageKey` and
+:php:`expectedLabels` as named arguments in the test:
+
+.. code-block:: php
+    :caption: PHPUnit data provider using named arguments
+
+    final class XliffParserTest extends UnitTestCase
+    {
+        public static function canParseXliffDataProvider(): \Generator
+        {
+            yield 'Can handle default' => [
+                'languageKey' => 'default',
+                'expectedLabels' => [
+                    'label1' => 'This is label #1',
+                    'label2' => 'This is label #2',
+                    'label3' => 'This is label #3',
+                ],
+            ];
+            yield 'Can handle translation' => [
+                'languageKey' => 'fr',
+                'expectedLabels' => [
+                    'label1' => 'Ceci est le libellé no. 1',
+                    'label2' => 'Ceci est le libellé no. 2',
+                    'label3' => 'Ceci est le libellé no. 3',
+                ],
+            ];
+        }
+
+        #[DataProvider('canParseXliffDataProvider')]
+        #[Test]
+        public function canParseXliff(string $languageKey, array $expectedLabels): void
+        {
+            // Test implementation
+        }
+    }
+
+Leveraging Named Arguments When Invoking PHP Functions
+------------------------------------------------------
+
+TYPO3 core may leverage named arguments when calling PHP functions, provided it
+enhances readability and simplifies the invocation. It is allowed for functions
+with more than three arguments. If named arguments are used, all arguments must
```

Comment:
So no "haystack" and "needle" allowed for inconsitent PHP functions? Should this explicitly be referenced, if really not allowed? At least for me that's a key benefit when deciding to allow named parameters...?

Reply:
not in method with only 2 arguments. we do *not* want to have named arguments for php functions used inflationary, just because dev can't remember implode() glue and array order. i think the rules phrasing tells that already.